OneOneOne/OneOneOne/message_handler.py

```diff
@@ -0,0 +1,190 @@
+from lxml import etree, objectify
+import xml.etree.ElementTree as ET
+
+from builder.pystache_message_builder import PystacheMessageBuilder
+from utilities.file_utilities import FileUtilities
+from definitions import XML_PATH, TEMPLATE_PATH
+import logging
+
+basic_success_response = FileUtilities.get_file_string(XML_PATH / 'basic_success_response.xml')
+
+base_fault_response = FileUtilities.get_file_string(XML_PATH / 'basic_fault_response.xml')
+
+
+def build_error_message(error):
+    builder = PystacheMessageBuilder(str(TEMPLATE_PATH), 'base_error_template')
+    return builder.build_message({"errorMessage": error})
+
+
+class MessageHandler:
+    namespaces = {
+        'soap': 'http://schemas.xmlsoap.org/soap/envelope/',
+        'a': 'http://www.etis.fskab.se/v1.0/ETISws',
+        'wsa': 'http://www.w3.org/2005/08/addressing',
+        'itk': 'urn:nhs-itk:ns:201005'
+    }
+
+    def __init__(self, message_string):
+        logging.basicConfig(level=logging.DEBUG)
+        self.message = message_string
+        self.message_tree = ET.fromstring(self.message)
+
+        self.check_list = [
+            self.check_action_types,
+            self.check_manifest_and_payload_count,
+            self.check_manifest_count_against_actual,
+            self.check_payload_count_against_actual,
+            self.check_payload_id_matches_manifest_id
+        ]
+
+    def evaluate_message(self):
+        """
+        This iterates over the message check methods searching for errors in the message, if no errors are found
+        a success response is returned
+
+        :return:
+        """
+        for check in self.check_list:
+            status, response = check()
+            if status != 200:
+                return status, response
+
+        return 200, basic_success_response
+
+    def check_action_types(self):
+        """
+        This method checks for equality between the action type in the header, and the service value in the message
+        body as per the 'DE_INVSER' requirement specified in the requirements spreadsheet
+        :return: status code, response content
+        """
+        action = ""
+        for type_tag in self.message_tree.findall("./soap:Header"
```

Comment:
I think perhaps naming this to `action_tag` and the one below to `service_tag` may make it clear that they are different types of tags you are iterating over.

Reply:
Yeah that's not greatly named, I've updated it to be `action_tag_value` to specify it's the actual value of the tag as well and not just the tag object.